tests/unit/v1/test_async_stream_iterator.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import sys
+
+
+def _make_async_stream_iterator(iterable):
+    from google.cloud.firestore_v1.async_stream_iterator import AsyncStreamIterator
+
+    async def _inner_generator():
+        for i in iterable:
+            X = yield i
+            if X:
+                yield X
+
+    return AsyncStreamIterator(_inner_generator())
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="requires python3.10 or higher")
```

Comment:
Because built-in `anext()` and `aiter()` were introduced in python 3.10 ([link](https://stackoverflow.com/a/73074973/8216546)). But on a second thought, I think I can call the dunder method directly, so we don't skip tests based on python version.